tests/generic_inline_admin/tests.py

```diff
@@ -429,3 +429,26 @@ class EpisodeAdmin(admin.ModelAdmin):
         inlines = ma.get_inline_instances(request)
         for (formset, inline), other_inline in zip(ma.get_formsets_with_inlines(request), inlines):
             self.assertIsInstance(formset, other_inline.get_formset(request).__class__)
+
+    def test_get_inline_instances_override_get_inlines(self):
+        """
+        get_formsets_with_inlines() returns the correct tuples.
+        """
+        class MediaForm(ModelForm):
+            class Meta:
+                model = Media
+                exclude = ['url']
+
+        class MediaInline(GenericTabularInline):
+            model = Media
+
+        class EpisodeAdmin(admin.ModelAdmin):
+            inlines = [
+                MediaInline
+            ]
+
+            def get_inlines(self, request, obj=None):
+                return []
+
+        ma = EpisodeAdmin(Episode, self.site)
+        self.assertEqual(ma.get_inline_instances(request), [])
```

Comment:
There should also be an assertion for the output of the new `get_inlines()` method.